internal/providers/openstack/openstack.go

```diff
@@ -60,9 +60,12 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 	}
 
 	var data []byte
+	errChan := make(chan error)
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	dispatchCount := 0
 
 	dispatch := func(name string, fn func() ([]byte, error)) {
+		dispatchCount++
```

Comment:
This isn't thread-safe.  And locking won't fix it either, since goroutine startup might be arbitrarily delayed.  I'd be inclined to move the goroutine launch inside dispatch(), after the increment.